boards/recovery/src/main.rs

```diff
@@ -136,6 +146,24 @@ mod app {
         loop {}
     }
 
+    // interrupt handler for recovery counter
+    #[task(binds=TC2, shared=[data_manager, recovery_timer])]
+    fn recovery_counter_tick(mut cx: recovery_counter_tick::Context) {
+        cx.shared.recovery_timer.lock(|timer| {
+            if timer.wait().is_ok() {
+                cx.shared.data_manager.lock(|data| {
+                    data.recovery_counter += 1;
+                });
+            }
+            // restart timer after interrupt
```

Comment:
This logic should be inside the if statement, not outside of it. This is because the timer interrupt can fire for multiple reasons and right now you are restarting the time for any and all interrupts, not just the case that it is finished.